handlers/geolocation/geolocation.go

```diff
@@ -26,29 +27,72 @@ type GeolocationHandlersCollection struct {
 
 // redirect an incoming user to a node for playback or 404 handling
 func (c *GeolocationHandlersCollection) RedirectHandler() httprouter.Handle {
-	return func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-		nodeHost := c.Config.NodeHost
-		redirectPrefixes := c.Config.RedirectPrefixes
+	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+		host := r.Host
+		prefix, playbackID, pathTmpl, isValid := parsePlaybackID(r.URL.Path)
+
+		if c.Config.CdnRedirectHost != "" {
+			hostNoPort, _, err := net.SplitHostPort(r.Host)
+			if err != nil {
+				hostNoPort = r.Host
+			}
+
+			if !isValid {
+				glog.Warningf("Can not parse playbackID from path %s", r.URL.Path)
+				w.WriteHeader(http.StatusNotFound)
+				return
+			}
+
+			if hostNoPort == c.Config.CdnRedirectHost {
+				glog.V(6).Info("Request comes from CDN. Returning playback directly")
+				newURL, err := url.Parse(r.URL.String())
+				if err != nil {
+					glog.Errorf("failed to parse incoming url for redirect url=%s err=%s", r.URL.String(), err)
+					w.WriteHeader(http.StatusInternalServerError)
+					return
+				}
+				newURL.Scheme = protocol(r)
+				newURL.Host = c.Config.NodeHost
```

Comment:
Not sure where should I redirect to avoid calling `Balancer.GetBestNode`?
`c.Config.NodeHost` makes sense but not sure if it was introduced for this purpose?
@iameli

Reply:
<img width="481" alt="Screenshot 2023-08-14 at 17 12 12" src="https://github.com/livepeer/catalyst-api/assets/3940860/f32514ac-9e93-42da-b4c4-e15dbc5834e4">